### PR TITLE
feat: add protein enrichment

### PR DESCRIPTION
**Pull request by gurdeep330**
# For authors

## Description
This PR introduces an example notebook demonstrating how users can leverage our utility functions to perform multimodal enrichment and embedding over a knowledge graph, using PrimeKG as a case study.

Specifically, this notebook showcases enrichment of gene/protein nodes with:
1. Protein sequences using ESM-2
2. Functional descriptions using BioBERT

This serves as the first in a series of PRs that will illustrate how to enrich and embed various biomedical entities in a KG, including:
1. GO terms
2. Reactome pathways
3. Disease ontologies
4. Cell ontologies
5. Drugs
6. More ....

## Type of change
Please delete options that are not relevant.
- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?
Please describe the tests you conducted to verify your changes. These may involve creating new test scripts or updating existing ones.
- [X] Added new test(s) in the ```tests``` folder
- [X] Added new function(s) to an existing test(s) (e.g.: ```tests/testX.py```)
- [ ] No new tests added (Please explain the rationale in this case)

## Checklist
- [X] My code follows the style guidelines mentioned in the Code/DevOps guides
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation (e.g. MkDocs)
- [X] My changes generate no new warnings
- [X] I have added or updated tests (in the ```tests``` folder) that prove my fix is effective or that my feature works
- [X] New and existing tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

# For reviewers

## Checklist pre-approval
- [ ] Is there enough documentation?  
- [ ] If a new feature has been added, or a bug fixed, has a test been added to confirm good behavior?
- [ ] Does the test(s) successfully test edge/corner cases?
- [ ] Does the PR pass the tests? (if the repository has continuous integration)

## Checklist post-approval
- [ ] Does this PR merge ```develop``` into ```main```? If so, please make sure to add a prefix (feat/fix/chore) and/or a suffix BREAKING CHANGE (if it's a major release) to your commit message.
- [ ] Does this PR close an issue? If so, please make sure to descriptively close this issue when the PR is merged.

## Checklist post-merge
- [ ] When you approve of the PR, merge and close it (Read this [article](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/about-merge-methods-on-github) to know about different merge methods on GitHub)
- [ ] Did this PR merge ```develop``` into ```main``` and is it suppose to run an automated release workflow (if applicable)? If so, please make sure to check under the "Actions" tab to see if the workflow has been initiated, and return later to verify that it has completed successfully.

### CODE REVIEW

File: aiagents4pharma/talk2knowledgegraphs/tests/test_utils_enrichments_uniprot.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python3
+
+"""
+Test cases for utils/enrichments/pubchem_strings.py
```
Comment (dmccloskey) [inline aiagents4pharma/talk2knowledgegraphs/tests/test_utils_enrichments_uniprot.py:4]:
```suggestion
Test cases for utils/enrichments/uniprot_proteins.py
```

File: aiagents4pharma/talk2knowledgegraphs/tests/test_utils_enrichments_uniprot.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import pytest
+from ..utils.enrichments.uniprot_proteins import EnrichmentWithUniProt
+
+# In this test, we will consider 2 examples:
+# 1. Gene Name: TP53
+# 2. Gene Name: TP5 (Incomplete; must return empty results)
+# 2. Gene Name: XZ (Shorter than 3 characters; must return empty results)
+# The expected description of TP53 startswith:
+START_DESCP = "Multifunctional transcription factor"
+# The expected description of TP53 startswith:
```
Comment (dmccloskey) [inline aiagents4pharma/talk2knowledgegraphs/tests/test_utils_enrichments_uniprot.py:16]:
```suggestion
# The expected amino acid sequence of TP53 startswith:
```

File: aiagents4pharma/talk2knowledgegraphs/tests/test_utils_enrichments_uniprot.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+START_SEQ = "MEEPQSDPSV"
+# The expected SMILES representation for the second PubChem ID is None.
```
Comment (dmccloskey) [inline aiagents4pharma/talk2knowledgegraphs/tests/test_utils_enrichments_uniprot.py:18]:
```suggestion
```

File: docs/notebooks/talk2knowledgegraphs/tutorial_primekg_enrichment.ipynb
Comment (dmccloskey) [inline docs/notebooks/talk2knowledgegraphs/tutorial_primekg_enrichment.ipynb:1]: The tutorial is self-explanatory 👍. There are quite a few typos, but this can be taken care of at a later time.

**Review by dmccloskey — APPROVED** [review]
This looks like a great starting point for adding protein enrichment and embeddings 🤩. A couple of very minor suggestions.

File: aiagents4pharma/talk2knowledgegraphs/tests/test_utils_enrichments_uniprot.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+@pytest.fixture(name="enrich_obj")
+def fixture_pubchem_config():
```
Comment (awmulyadi) [inline aiagents4pharma/talk2knowledgegraphs/tests/test_utils_enrichments_uniprot.py:21]: As of now, some method and variable names still refer to PubChem instead of UniProt/gene names.

File: aiagents4pharma/talk2knowledgegraphs/tests/test_utils_enrichments_uniprot.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    """Return a dictionary with the configuration for UniProt enrichment."""
+    return EnrichmentWithUniProt()
+
+def test_enrich_documents(enrich_obj):
+    """Test the enrich_documents method."""
+    pubchem_ids = ["TP53", "TP5", "XZ"]
```
Comment (awmulyadi) [inline aiagents4pharma/talk2knowledgegraphs/tests/test_utils_enrichments_uniprot.py:27]: I believe this list should refer to the gene names to be enriched.

File: aiagents4pharma/talk2knowledgegraphs/tests/test_utils_enrichments_uniprot.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    assert descriptions[0].startswith(START_DESCP)
+    assert sequences[0].startswith(START_SEQ)
+    assert descriptions[1] is None
+    assert sequences[1] is None
+    assert descriptions[2] is None
+    assert sequences[2] is None
+
+def test_enrich_documents_with_rag(enrich_obj):
+    """Test the enrich_documents_with_rag method."""
+    pubchem_ids = ["TP53", "TP5", "XZ"]
```
Comment (awmulyadi) [inline aiagents4pharma/talk2knowledgegraphs/tests/test_utils_enrichments_uniprot.py:38]: This list as well.

File: aiagents4pharma/talk2knowledgegraphs/configs/utils/enrichments/uniprot_proteins/default.yaml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+_target_: talk2knowledgegraphs.utils.enrichments.uniprot_proteins
+uniprot_url: https://www.ebi.ac.uk/proteins/api/proteins
+reviewed: true
+isoform: 0
+organism: Homo sapiens
```
Comment (awmulyadi) [inline aiagents4pharma/talk2knowledgegraphs/configs/utils/enrichments/uniprot_proteins/default.yaml:5]: Other than this default organism, where can we find available organisms as an option? I am asking because perhaps the user might be interested in other organisms for their use case.

File: aiagents4pharma/talk2knowledgegraphs/utils/enrichments/uniprot_proteins.py [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+                    for value in comment['text']:
+                        description += value['value']
+            sequence = response_body[0]['sequence']['sequence']
+            descriptions.append(description)
+            sequences.append(sequence)
+        return descriptions, sequences
+
+    def enrich_documents_with_rag(self, texts, docs):
+        """
+        Enrich a list of input PubChem IDs with their STRINGS representation.
```
Comment (awmulyadi) [inline aiagents4pharma/talk2knowledgegraphs/utils/enrichments/uniprot_proteins.py:79]: The following comments for the documentation still refer to PubChem IDs.

**Review by awmulyadi — APPROVED** [review]
Hi Gurdeep. This is a nice update on enrichment over the PrimeKG using UniProt. I just have small, minor comments regarding naming and documentation. 👍

**Comment by gurdeep330** [comment]
Thank you both for the suggestions 🙏
